### PR TITLE
Publicly usable lint checks

### PR DESCRIPTION
**Pull request by samtstern**
See #1351

I wonder if we should create an empty `firebaseui-lint` android library that only exists to have the `lintChecks` call.  I say that because I am not sure what happens if we had multiple (for example auth and firestore) depending on the `lint` module as `lintChecks`

### CODE REVIEW

**Comment by samtstern** [comment]
cc @pavlospt wdyt?

**Comment by samtstern** [comment]
Remaining error:
```
Errors found:
  
  /usr/local/google/home/samstern/Projects/firebase/firebaseui-android/auth/src/main/res/values/com_crashlytics_export_strings.xml:9: Error: Resource named 'com.crashlytics.android.build_id' does not start with the project's resource prefix 'fui_'; rename to 'fui_com.crashlytics.android.build_id' ? [ResourceName]
      <string tools:ignore="UnusedResources,TypographyDashes" name="com.crashlytics.android.build_id" translatable="false">77252c51-646a-4395-affc-28655847aec9</string>
```

I know we hit this before but I forget why.

**Comment by pavlospt** [comment]
If there was a firebaseui-lint library it could be served as a dependency as well for standalone usage! Otherwise the separation looks good to me!

File: proguard-tests/build.gradle.kts [hunk 1]
```diff
@@ -4,6 +4,14 @@ android {
     }
 
     buildTypes {
+        named("debug").configure {
```
Comment (SUPERCILEX) [inline proguard-tests/build.gradle.kts]: Why'd we add this back? It adds like 5+ mins to a full debug build. Shouldn't we only run it in CI?

Comment (samtstern) [reply proguard-tests/build.gradle.kts]: Oh damn I forgot why we removed it.  Android Studio gets pretty angry when not all modules have matching configurations and this made it happy.

Let me see if I can appease Android Studio without compromising time.

Comment (SUPERCILEX) [reply proguard-tests/build.gradle.kts]: Ohhh, yeah, I remember seeing those warnings. Let's use a variant filter then: https://developer.android.com/studio/build/build-variants#filter-variants

Comment (samtstern) [reply proguard-tests/build.gradle.kts]: Yep it looks like that's already working with `setIgnore`.  I pushed a commit to simplify the `debug` config to the minimum that makes AS happy.

I did a `time` of some clean builds and the extra config doesn't cost us anything.  Will confirm via Travis logs.

Comment (SUPERCILEX) [reply proguard-tests/build.gradle.kts]: Awesome, SGTM!

Comment (samtstern) [reply proguard-tests/build.gradle.kts]: Travis confirmed: no slowdown.

**Pull request merged**